### ISSUE TITLE
[BUG] Full hyper search casue Nan about metric.

### ISSUE THREAD

**Issue by LilaKen** [Secure-and-Intelligent-Systems-Lab/WPMixer#7]
**full_hyperSearch_result_Weather_96.log**
Args in experiment: Namespace(model='WPMixer', task_name='long_term_forecast', data='Weather', use_hyperParam_optim=False, no_decomposition=False, use_multi_gpu=False, n_jobs=1, seed=42, seq_len=512, pred_len=96, d_model=256, tfactor=3, dfactor=7, wavelet='db3', level=2, patch_len=16, stride=8, batch_size=32, learning_rate=0.000913333, dropout=0.4, embedding_dropout=0.1, weight_decay=0.0, patience=12, train_epochs=60, label_len=0, seasonal_patterns='Monthly', features='M', target='OT', freq='h', checkpoints='./checkpoints/', cols=None, num_workers=0, itr=1, lradj='type3', use_amp=True, use_gpu=True, gpu=0, devices='0,1', embed=0, loss='smoothL1', pct_start=0.2, optuna_seq_len=None, optuna_lr=None, optuna_batch=None, optuna_wavelet=None, optuna_tfactor=None, optuna_dfactor=None, optuna_epochs=None, optuna_dropout=None, optuna_embedding_dropout=None, optuna_patch_len=None, optuna_stride=None, optuna_lradj=None, optuna_dmodel=None, optuna_weight_decay=None, optuna_patience=None, optuna_level=None, optuna_trial_num=None, data_path='weather.csv', root_path='./data/weather/', c_in=21, c_out=21, detail_freq='h')
Use GPU: cuda:0
Start Training- WPMixer_Weather_dec-True_sl512_pl96_dm256_bt32_wvdb3_tf3_df7_ptl16_stl8_sd42
train 36280
val 5175
test 10444
Epoch 1: cost time: 44.69 sec
	Epoch 1: Steps- 1133 | Train Loss: 0.11334 Vali.MSE: 0.38644 Vali.MAE: 0.26358 Test.MSE: 0.14351 Test.MAE: 0.18933
	Validation loss decreased (inf --> 0.386441).  Saving model ...
Updating learning rate to 0.000913333
Epoch 2: cost time: 44.57 sec
	Epoch 2: Steps- 1133 | Train Loss: 0.10096 Vali.MSE: 0.38166 Vali.MAE: 0.25908 Test.MSE: 0.14350 Test.MAE: 0.18938
	Validation loss decreased (0.386441 --> 0.381664).  Saving model ...
Updating learning rate to 0.000913333
Epoch 3: cost time: 49.31 sec
	Epoch 3: Steps- 1133 | Train Loss: 0.09836 Vali.MSE: 0.38948 Vali.MAE: 0.26755 Test.MSE: 0.14309 Test.MAE: 0.19003
	EarlyStopping counter: 1 out of 12
Updating learning rate to 0.000913333
Epoch 4: cost time: 46.51 sec
	Epoch 4: Steps- 1133 | Train Loss: nan Vali.MSE: nan Vali.MAE: nan Test.MSE: nan Test.MAE: nan
	Validation loss decreased (0.381664 --> nan).  Saving model ...
\stopping: train-loss-nan
Start Testing- WPMixer_Weather_dec-True_sl512_pl96_dm256_bt32_wvdb3_tf3_df7_ptl16_stl8_sd42
test 10444
mse: nan, mae: nan

**full_hyperSearch_result_Weather_720.log**
Args in experiment: Namespace(model='WPMixer', task_name='long_term_forecast', data='Weather', use_hyperParam_optim=False, no_decomposition=False, use_multi_gpu=False, n_jobs=1, seed=42, seq_len=512, pred_len=720, d_model=128, tfactor=7, dfactor=5, wavelet='db2', level=1, patch_len=16, stride=8, batch_size=128, learning_rate=0.001470479, dropout=0.4, embedding_dropout=0.2, weight_decay=0.0, patience=12, train_epochs=60, label_len=0, seasonal_patterns='Monthly', features='M', target='OT', freq='h', checkpoints='./checkpoints/', cols=None, num_workers=0, itr=1, lradj='type3', use_amp=True, use_gpu=True, gpu=0, devices='0,1', embed=0, loss='smoothL1', pct_start=0.2, optuna_seq_len=None, optuna_lr=None, optuna_batch=None, optuna_wavelet=None, optuna_tfactor=None, optuna_dfactor=None, optuna_epochs=None, optuna_dropout=None, optuna_embedding_dropout=None, optuna_patch_len=None, optuna_stride=None, optuna_lradj=None, optuna_dmodel=None, optuna_weight_decay=None, optuna_patience=None, optuna_level=None, optuna_trial_num=None, data_path='weather.csv', root_path='./data/weather/', c_in=21, c_out=21, detail_freq='h')
Use GPU: cuda:0
Start Training- WPMixer_Weather_dec-True_sl512_pl720_dm128_bt128_wvdb2_tf7_df5_ptl16_stl8_sd42
train 35656
val 4551
test 9820
Epoch 1: cost time: 18.09 sec
	Epoch 1: Steps- 278 | Train Loss: 0.18708 Vali.MSE: 0.62162 Vali.MAE: 0.41583 Test.MSE: 0.30740 Test.MAE: 0.32140
	Validation loss decreased (inf --> 0.621624).  Saving model ...
Updating learning rate to 0.001470479
Epoch 2: cost time: 17.53 sec
	Epoch 2: Steps- 278 | Train Loss: 0.17421 Vali.MSE: 0.59951 Vali.MAE: 0.40807 Test.MSE: 0.31543 Test.MAE: 0.32832
	Validation loss decreased (0.621624 --> 0.599511).  Saving model ...
Updating learning rate to 0.001470479
Epoch 3: cost time: 17.45 sec
	Epoch 3: Steps- 278 | Train Loss: 0.16647 Vali.MSE: 0.61381 Vali.MAE: 0.41475 Test.MSE: 0.32784 Test.MAE: 0.33476
	EarlyStopping counter: 1 out of 12
Updating learning rate to 0.001470479
Epoch 4: cost time: 17.30 sec
	Epoch 4: Steps- 278 | Train Loss: 0.15758 Vali.MSE: 0.62675 Vali.MAE: 0.42429 Test.MSE: 0.33828 Test.MAE: 0.34131
	EarlyStopping counter: 2 out of 12
Updating learning rate to 0.0013234311
Epoch 5: cost time: 16.86 sec
	Epoch 5: Steps- 278 | Train Loss: 0.14921 Vali.MSE: 0.63936 Vali.MAE: 0.42890 Test.MSE: 0.34707 Test.MAE: 0.34681
	EarlyStopping counter: 3 out of 12
Updating learning rate to 0.0011910879900000001
Epoch 6: cost time: 17.61 sec
	Epoch 6: Steps- 278 | Train Loss: 0.14370 Vali.MSE: 0.64226 Vali.MAE: 0.43084 Test.MSE: 0.34878 Test.MAE: 0.34808
	EarlyStopping counter: 4 out of 12
Updating learning rate to 0.0010719791910000001
Epoch 7: cost time: 17.16 sec
	Epoch 7: Steps- 278 | Train Loss: 0.13990 Vali.MSE: 0.63826 Vali.MAE: 0.43193 Test.MSE: 0.35195 Test.MAE: 0.34960
	EarlyStopping counter: 5 out of 12
Updating learning rate to 0.0009647812719000001
Epoch 8: cost time: 17.18 sec
	Epoch 8: Steps- 278 | Train Loss: 0.13686 Vali.MSE: 0.64560 Vali.MAE: 0.43319 Test.MSE: 0.35237 Test.MAE: 0.34851
	EarlyStopping counter: 6 out of 12
Updating learning rate to 0.0008683031447100001
Epoch 9: cost time: 17.02 sec
	Epoch 9: Steps- 278 | Train Loss: 0.13459 Vali.MSE: 0.65252 Vali.MAE: 0.43705 Test.MSE: 0.35448 Test.MAE: 0.34946
	EarlyStopping counter: 7 out of 12
Updating learning rate to 0.0007814728302390001
Epoch 10: cost time: 17.73 sec
	Epoch 10: Steps- 278 | Train Loss: nan Vali.MSE: nan Vali.MAE: nan Test.MSE: nan Test.MAE: nan
	Validation loss decreased (0.599511 --> nan).  Saving model ...
\stopping: train-loss-nan
Start Testing- WPMixer_Weather_dec-True_sl512_pl720_dm128_bt128_wvdb2_tf7_df5_ptl16_stl8_sd42
test 9820
mse: nan, mae: nan

Some code will cause this problem as above, I didn't try all the file.

**Comment by zhujiayi18161778290** [comment]
Args in experiment: Namespace(model='WPMixer', task_name='long_term_forecast', data='Weather', use_hyperParam_optim=False, no_decomposition=False, use_multi_gpu=False, n_jobs=1, seed=42, seq_len=96, pred_len=192, d_model=256, tfactor=5, dfactor=3, wavelet='coif4', level=1, patch_len=16, stride=8, batch_size=128, learning_rate=0.003129047, dropout=0.4, embedding_dropout=0.2, weight_decay=0.0, patience=5, train_epochs=20, label_len=0, seasonal_patterns='Monthly', features='M', target='OT', freq='h', checkpoints='./checkpoints/', cols=None, num_workers=0, itr=1, lradj='type3', use_amp=True, use_gpu=True, gpu=0, devices='0,1', embed=0, loss='smoothL1', pct_start=0.2, optuna_seq_len=None, optuna_lr=None, optuna_batch=None, optuna_wavelet=None, optuna_tfactor=None, optuna_dfactor=None, optuna_epochs=None, optuna_dropout=None, optuna_embedding_dropout=None, optuna_patch_len=None, optuna_stride=None, optuna_lradj=None, optuna_dmodel=None, optuna_weight_decay=None, optuna_patience=None, optuna_level=None, optuna_trial_num=None, data_path='weather.csv', root_path='./data/weather/', c_in=21, c_out=21, detail_freq='h')
Use GPU: cuda:0
Start Training- WPMixer_Weather_dec-True_sl96_pl192_dm256_bt128_wvcoif4_tf5_df3_ptl16_stl8_sd42
train 36600
val 5079
test 10348
Epoch 1: cost time: 87.70 sec
	Epoch 1: Steps- 285 | Train Loss: 0.15526 Vali.MSE: 0.47944 Vali.MAE: 0.31703 Test.MSE: 0.21199 Test.MAE: 0.24899
	Validation loss decreased (inf --> 0.479443).  Saving model ...
Updating learning rate to 0.003129047
Epoch 2: cost time: 81.22 sec
	Epoch 2: Steps- 285 | Train Loss: 0.14499 Vali.MSE: 0.47166 Vali.MAE: 0.31088 Test.MSE: 0.20969 Test.MAE: 0.24573
	Validation loss decreased (0.479443 --> 0.471658).  Saving model ...
Updating learning rate to 0.003129047
Epoch 3: cost time: 77.44 sec
	Epoch 3: Steps- 285 | Train Loss: 0.14347 Vali.MSE: 0.47274 Vali.MAE: 0.31353 Test.MSE: 0.21167 Test.MAE: 0.24887
	EarlyStopping counter: 1 out of 5
Updating learning rate to 0.003129047
Epoch 4: cost time: 71.67 sec
	Epoch 4: Steps- 285 | Train Loss: 0.14268 Vali.MSE: 0.46900 Vali.MAE: 0.31096 Test.MSE: 0.20861 Test.MAE: 0.24558
	Validation loss decreased (0.471658 --> 0.469005).  Saving model ...
Updating learning rate to 0.0028161423
Epoch 5: cost time: 70.33 sec
	Epoch 5: Steps- 285 | Train Loss: 0.14196 Vali.MSE: 0.46755 Vali.MAE: 0.30973 Test.MSE: 0.20935 Test.MAE: 0.24624
	Validation loss decreased (0.469005 --> 0.467546).  Saving model ...
Updating learning rate to 0.00253452807
Epoch 6: cost time: 71.67 sec
	Epoch 6: Steps- 285 | Train Loss: 0.14127 Vali.MSE: 0.46710 Vali.MAE: 0.31025 Test.MSE: 0.20973 Test.MAE: 0.24661
	Validation loss decreased (0.467546 --> 0.467099).  Saving model ...
Updating learning rate to 0.0022810752630000003
Epoch 7: cost time: 74.29 sec
	Epoch 7: Steps- 285 | Train Loss: 0.14082 Vali.MSE: 0.46996 Vali.MAE: 0.30979 Test.MSE: 0.20899 Test.MAE: 0.24520
	EarlyStopping counter: 1 out of 5
Updating learning rate to 0.0020529677366999998
Epoch 8: cost time: 59.86 sec
	Epoch 8: Steps- 285 | Train Loss: nan Vali.MSE: nan Vali.MAE: nan Test.MSE: nan Test.MAE: nan
	Validation loss decreased (0.467099 --> nan).  Saving model ...
\stopping: train-loss-nan
Start Testing- WPMixer_Weather_dec-True_sl96_pl192_dm256_bt128_wvcoif4_tf5_df3_ptl16_stl8_sd42
test 10348
mse: nan, mae: nan




Args in experiment: Namespace(model='WPMixer', task_name='long_term_forecast', data='Weather', use_hyperParam_optim=False, no_decomposition=False, use_multi_gpu=False, n_jobs=1, seed=42, seq_len=96, pred_len=336, d_model=128, tfactor=7, dfactor=7, wavelet='coif4', level=1, patch_len=16, stride=8, batch_size=128, learning_rate=0.004176281, dropout=0.4, embedding_dropout=0.1, weight_decay=0.0, patience=5, train_epochs=20, label_len=0, seasonal_patterns='Monthly', features='M', target='OT', freq='h', checkpoints='./checkpoints/', cols=None, num_workers=0, itr=1, lradj='type3', use_amp=True, use_gpu=True, gpu=0, devices='0,1', embed=0, loss='smoothL1', pct_start=0.2, optuna_seq_len=None, optuna_lr=None, optuna_batch=None, optuna_wavelet=None, optuna_tfactor=None, optuna_dfactor=None, optuna_epochs=None, optuna_dropout=None, optuna_embedding_dropout=None, optuna_patch_len=None, optuna_stride=None, optuna_lradj=None, optuna_dmodel=None, optuna_weight_decay=None, optuna_patience=None, optuna_level=None, optuna_trial_num=None, data_path='weather.csv', root_path='./data/weather/', c_in=21, c_out=21, detail_freq='h')
Use GPU: cuda:0
Start Training- WPMixer_Weather_dec-True_sl96_pl336_dm128_bt128_wvcoif4_tf7_df7_ptl16_stl8_sd42
train 36456
val 4935
test 10204
Epoch 1: cost time: 43.07 sec
	Epoch 1: Steps- 284 | Train Loss: 0.18204 Vali.MSE: 0.56422 Vali.MAE: 0.37603 Test.MSE: 0.26697 Test.MAE: 0.29097
	Validation loss decreased (inf --> 0.564224).  Saving model ...
Updating learning rate to 0.004176281
Epoch 2: cost time: 41.49 sec
	Epoch 2: Steps- 284 | Train Loss: 0.17427 Vali.MSE: 0.55862 Vali.MAE: 0.37164 Test.MSE: 0.26700 Test.MAE: 0.29082
	Validation loss decreased (0.564224 --> 0.558619).  Saving model ...
Updating learning rate to 0.004176281
Epoch 3: cost time: 40.87 sec
	Epoch 3: Steps- 284 | Train Loss: 0.17284 Vali.MSE: 0.55572 Vali.MAE: 0.36832 Test.MSE: 0.26453 Test.MAE: 0.28735
	Validation loss decreased (0.558619 --> 0.555720).  Saving model ...
Updating learning rate to 0.004176281
Epoch 4: cost time: 35.82 sec
	Epoch 4: Steps- 284 | Train Loss: 0.17214 Vali.MSE: 0.55160 Vali.MAE: 0.36929 Test.MSE: 0.26419 Test.MAE: 0.28839
	Validation loss decreased (0.555720 --> 0.551596).  Saving model ...
Updating learning rate to 0.0037586529
Epoch 5: cost time: 30.32 sec
	Epoch 5: Steps- 284 | Train Loss: 0.17117 Vali.MSE: 0.55234 Vali.MAE: 0.37048 Test.MSE: 0.26374 Test.MAE: 0.28877
	EarlyStopping counter: 1 out of 5
Updating learning rate to 0.0033827876099999997
Epoch 6: cost time: 30.40 sec
	Epoch 6: Steps- 284 | Train Loss: 0.17062 Vali.MSE: 0.54861 Vali.MAE: 0.36724 Test.MSE: 0.26382 Test.MAE: 0.28854
	Validation loss decreased (0.551596 --> 0.548609).  Saving model ...
Updating learning rate to 0.003044508849
Epoch 7: cost time: 30.41 sec
	Epoch 7: Steps- 284 | Train Loss: 0.17015 Vali.MSE: 0.54768 Vali.MAE: 0.36431 Test.MSE: 0.26339 Test.MAE: 0.28676
	Validation loss decreased (0.548609 --> 0.547679).  Saving model ...
Updating learning rate to 0.0027400579640999997
Epoch 8: cost time: 30.37 sec
	Epoch 8: Steps- 284 | Train Loss: 0.16974 Vali.MSE: 0.54579 Vali.MAE: 0.36791 Test.MSE: 0.26340 Test.MAE: 0.28862
	Validation loss decreased (0.547679 --> 0.545786).  Saving model ...
Updating learning rate to 0.00246605216769
Epoch 9: cost time: 30.36 sec
	Epoch 9: Steps- 284 | Train Loss: nan Vali.MSE: nan Vali.MAE: nan Test.MSE: nan Test.MAE: nan
	Validation loss decreased (0.545786 --> nan).  Saving model ...
\stopping: train-loss-nan
Start Testing- WPMixer_Weather_dec-True_sl96_pl336_dm128_bt128_wvcoif4_tf7_df7_ptl16_stl8_sd42
test 10204
mse: nan, mae: nan
